tests/mlmodel_langchain/test_tool.py

```diff
@@ -0,0 +1,242 @@
+    """A test tool that returns query string"""
+    return query
+
+
+@tool
+def multi_arg_tool(first_num: int, second_num: int):
+    """A test tool that adds two integers together"""
+    return first_num + second_num
+
+
+single_arg_tool_recorded_events = [
+    (
+        {"type": "LlmTool"},
+        {
+            "id": None,  # UUID that varies with each run
+            "appName": "Python Agent Test (mlmodel_langchain)",
+            "run_id": None,
+            "output": "Python Agent",
+            "name": "single_arg_tool",
+            "description": "single_arg_tool(query: str) - A test tool that returns query string",
+            "transaction_id": "transaction-id",
+            "span_id": None,
+            "trace_id": "trace-id",
+            "input": "{'query': 'Python Agent'}",
+            "vendor": "langchain",
+            "ingest_source": "Python",
+            "duration": None,
+            "tags": "",
+        },
+    ),
+]
+
+
+@reset_core_stats_engine()
+@validate_custom_events(single_arg_tool_recorded_events)
+@validate_custom_event_count(count=1)
+@validate_transaction_metrics(
+    name="test_tool:test_langchain_single_arg_tool",
+    scoped_metrics=[("Llm/tool/Langchain/run", 1)],
+    rollup_metrics=[("Llm/tool/Langchain/run", 1)],
+    custom_metrics=[
+        ("Python/ML/Langchain/%s" % langchain.__version__, 1),
+    ],
+    background_task=True,
+)
+@background_task()
+def test_langchain_single_arg_tool(set_trace_info):
+    set_trace_info()
+    single_arg_tool.run({"query": "Python Agent"})
+
+
+multi_arg_tool_recorded_events = [
+    (
+        {"type": "LlmTool"},
+        {
+            "id": None,  # UUID that varies with each run
+            "appName": "Python Agent Test (mlmodel_langchain)",
+            "run_id": None,
+            "output": "81",
+            "name": "multi_arg_tool",
+            "description": "multi_arg_tool(first_num: int, second_num: int) - A test tool that adds two integers together",
+            "transaction_id": "transaction-id",
+            "span_id": None,
+            "trace_id": "trace-id",
+            "input": "{'first_num': 53, 'second_num': 28}",
+            "vendor": "langchain",
+            "ingest_source": "Python",
+            "duration": None,
+            "tags": "['test_tags', 'python']",
+            "metadata.test": "langchain",
+            "metadata.test_run": True,
+        },
+    ),
+]
+
+
+@reset_core_stats_engine()
+@validate_custom_events(multi_arg_tool_recorded_events)
+@validate_custom_event_count(count=1)
+@validate_transaction_metrics(
+    name="test_tool:test_langchain_multi_arg_tool",
+    scoped_metrics=[("Llm/tool/Langchain/run", 1)],
+    rollup_metrics=[("Llm/tool/Langchain/run", 1)],
+    custom_metrics=[
+        ("Python/ML/Langchain/%s" % langchain.__version__, 1),
+    ],
+    background_task=True,
+)
+@background_task()
+def test_langchain_multi_arg_tool(set_trace_info):
+    set_trace_info()
+    multi_arg_tool.run(
```

Comment:
I think we could probably just combine this all into one test for the metadata and tags and then delete `test_langchain_tags_and_metadata_on_instance`:
```suggestion
    multi_arg_tool.metadata = {"test_run": True}
    multi_arg_tool.tags=["test_tags"]
    multi_arg_tool.run(
```